crates/ruff/src/checkers/imports.rs

```diff
@@ -54,5 +55,45 @@ pub fn check_imports(
             &blocks, python_ast, locator, stylist, settings, autofix,
         ));
     }
-    diagnostics
+    let mut imports: FxHashMap<PathBuf, Vec<Import>> = FxHashMap::default();
+    let mut imports_vec = vec![];
+    for &block in &blocks {
+        block.imports.iter().for_each(|&stmt| match &stmt.node {
+            StmtKind::Import { names } => {
+                // from testing, seems this should only have one entry
+                imports_vec.push(Import {
+                    name: names[0].node.name.clone(),
+                    location: stmt.location,
+                    end_location: stmt.end_location.unwrap(),
+                });
+            }
+            StmtKind::ImportFrom { module, names, .. } => imports_vec.extend(
+                names
+                    .iter()
+                    .map(|name| Import {
+                        name: format!(
+                            "{}.{}",
+                            {
+                                if let Some(n) = module {
+                                    n
+                                } else {
+                                    ""
+                                }
+                            },
+                            name.node.name
+                        ),
+                        location: name.location,
+                        end_location: name.end_location.unwrap(),
+                    })
+                    .collect::<Vec<Import>>(),
```

Comment:
Is it necessary to call `collect` here? `extend` should be fine with an `Iter` argument
```suggestion
```

Reply:
I'll have a look

Reply:
Forgive me for being a noob, but if we don't return a `Vec` here, won't this get a bit more complicated as then the hashmap has to contain something which will `impl` a trait? As a dynamically sized object, at least when I try to do this, I get the complaint that the size needs to be known at compile time.
Unless you mean that you'd like that, and I need to `Box` it?